types/matches.d.ts

```diff
@@ -1,6 +1,145 @@
 export type MatcherFunction = (content: string, element: HTMLElement) => boolean
 export type Matcher = string | RegExp | MatcherFunction
 
+// important to use String here
+// this gives us intellisense but it can also accept values
+// that are not included in the union types
+export type ByRoleMatcher = AllowedRoles | String | RegExp | MatcherFunction
+export type AllowedRoles =
```

Comment:
Should we include a link for the source of these so if there are new ones added we know where to go to get them?

Reply:
We should leverage `@types/aria-query` instead. This would automatically sync the types with the supported version of aria.

Reply:
Though we might want to wait for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45657.

Someone else could re-open that one and add themselves as an owner. Then we can fast track it as soon as I approve it 😄

Reply:
Great, thanks @eps1lon .

Reply:
Would still help if someone could review DefinitelyTyped/DefinitelyTyped#45657 since we'll be using those changes in the end.

Reply:
`@types/aria-query@^4.2.0` should have all the roles for aria 1.2